packages/ui-kit/lib/components/video-player.tsx

```diff
@@ -0,0 +1,164 @@
+            setIsLoading(true);
+        }
+        
+        // Clear any existing timeout when ID changes
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+        
+        // Set timeout to show error if video doesn't load within the specified time
+        // This helps prevent prolonged flickering
+        if (videoId) {
+            timeoutRef.current = setTimeout(() => {
+                if (isLoading) {
+                    setVideoError(true);
+                    setIsLoading(false);
+                    onErrorCallback(dictionary.components.videoPlayer.videoErrorText, new Error('Video load timeout'));
+                }
+            }, loadTimer);
+        }
+        
+        return () => {
+            if (timeoutRef.current) {
+                clearTimeout(timeoutRef.current);
+            }
+        };
+    }, [videoId, thumbnailUrl, dictionary, onErrorCallback, loadTimer]);
+
+    const handleThumbnailClick = () => {
+        setShowPlayer(true);
+        setAutoPlay(true);
+    };
+
+    const handleVideoError = (event: any) => {
+        setVideoError(true);
+        setIsLoading(false);
+        onErrorCallback(dictionary.components.videoPlayer.videoErrorText, event);
+        
+        // Clear timeout since we got a definitive error
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+    };
+    
+    const handlePlayerLoaded = () => {
+        setIsLoading(false);
+        
+        // Clear timeout since player loaded successfully
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+    };
+
+    // Error component
+    const ErrorDisplay = () => (
+        <div className="rounded-medium w-full min-w-[18rem] h-[16rem] bg-base-neutral-700 flex items-center justify-center p-4">
+            <span className="text-text-secondary text-md">
+                {dictionary.components.videoPlayer.videoErrorText}
+            </span>
+        </div>
+    );
+
+    // Loading component - shown while waiting for player to initialize
+    const LoadingDisplay = () => (
+      <div className='h-full w-full items-center justify-center flex'>
+        <div className="w-6 h-6 border-4 border-base-neutral-700 border-t-transparent rounded-full animate-spin"></div>
+        </div>
+    );
+
+    const playerStyle = videoError ? { display: 'none' } : {};
+
+    return (
+        <div className="w-full overflow-hidden">
+            {!showPlayer && thumbnailUrl ? (
+                <img
+                    src={thumbnailUrl}
+                    alt="Video Thumbnail"
+                    className="w-auto max-w-full h-auto object-contain cursor-pointer"
+                    onClick={handleThumbnailClick}
+                    onError={() => {
+                        if (!videoError) {
+                            setShowPlayer(true);
+                        }
+                    }}
+                />
```

Comment:
A minor thing a noticed is thumbnail's alt text being show for a fraction of a second if there's an error, similar to the flicker case with the MuxPlayer described above. Could this be handled similarly with display property relying on something like isImageLoading? Then this thumbnail component could be extracted outside VideoPlayer for better readability.